### PR TITLE
Add changesetIndex in addition to changesetId to IModelDb.

### PR DESCRIPTION
**Pull request by kabentley**
Of course since old versions didn't save it, it may be `undefined`. 

Relies on [this PR](https://bentleycs.visualstudio.com/iModelTechnologies/_git/imodel02/pullrequest/174373)

### CODE REVIEW

**Review by pmconne — CHANGES_REQUESTED** [review]
lint, extract-api, rush change.

**Comment by kabentley** [comment]
> lint, extract-api, rush change.

All done. Working on NextVersion

File: common/api/summary/imodeljs-backend.exports.csv [hunk 1]
```diff
@@ -40,15 +40,18 @@ public;CategoryOwnsSubCategories
 public;CategorySelector 
 internal;ChangedElementsDb 
 internal;ChangesetArg 
+<<<<<<< HEAD
```
Comment (pmconne) [inline common/api/summary/imodeljs-backend.exports.csv:43]: Fix merge conflict.

**Review by pmconne — CHANGES_REQUESTED** [review]
Merge conflicts in imodeljs-backend.exports.csv.

**Review by pmconne — APPROVED** [review]

**Review by bsteinbk — APPROVED** [review]

**Comment by pmconne** [comment]
Please merge ASAP - native PR already merged.

**Pull request merged**